e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/filter-payment-details.spec.js

```diff
@@ -26,52 +28,693 @@ context(`users can filter payment details by facility id and payment reference a
+        );
+
+        pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput().should('have.value', completeFacilityIdFilter);
+
+        pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(firstPayment.id, firstFeeRecord.id).should('exist');
+        pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(secondPayment.id, secondFeeRecord.id).should('not.exist');
+      });
+    });
+
+    describe('when a partial facility id filter is submitted', () => {
+      it('should only display the payments which partially match the supplied facility id and persist the inputted value', () => {
+        const partialFacilityIdFilter = '1111';
+
+        const firstFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).withFacilityId('77771111').build();
+        const secondFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(2).withFacilityId('22222222').build();
+
+        const firstPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+          .withId(11)
+          .withAmount(100)
+          .withReference('ABCD')
+          .withFeeRecords([firstFeeRecord])
+          .build();
+        const secondPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+          .withId(12)
+          .withAmount(200)
+          .withReference('EFGH')
+          .withFeeRecords([secondFeeRecord])
+          .build();
+
+        cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+        cy.task(NODE_TASKS.INSERT_PAYMENTS_INTO_DB, [firstPayment, secondPayment]);
+
+        pages.landingPage.visit();
+        cy.login(USERS.PDC_RECONCILE);
+
+        cy.visit(`/utilisation-reports/${reportId}`);
+
+        pages.utilisationReportPage.paymentDetailsTabLink().click();
+
+        cy.keyboardInput(pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput(), partialFacilityIdFilter);
+        pages.utilisationReportPage.paymentDetailsTab.filters.submitButton().click();
+
+        cy.url().should(
+          'eq',
+          relative(`/utilisation-reports/${reportId}?paymentDetailsPaymentReference=&paymentDetailsFacilityId=${partialFacilityIdFilter}#payment-details`),
+        );
+
+        pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput().should('have.value', partialFacilityIdFilter);
+
+        pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(firstPayment.id, firstFeeRecord.id).should('exist');
+        pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(secondPayment.id, secondFeeRecord.id).should('not.exist');
+      });
+    });
+
+    describe('when the facility id filter is submitted with an invalid value', () => {
+      it('should display an error message for invalid facility id and persist the inputted value', () => {
+        const invalidFacilityIdFilter = '123'; // Invalid due to length.
```

Comment:
Are we able to use a constant here e.g MIN - 1? I think we have a "minimum" constant somewhere already

Reply:
This one uses an existing regex which includes the length check, but I've updated the other invalid filter value for payment reference to use the `MIN_` constant that was introduced

Reply:
thanks @sw-mattw i'm not following though, i'm thinking something like this?

```
// const invalidFacilityIdFilter = '123';

const invalidFacilityIdFilter = MIN + 1;
```

Reply:
Apologies, the minimum constant that was recently introduced was for the payment reference filter (`MIN_PAYMENT_REFERENCE_FILTER_CHARACTER_COUNT`) rather than the facility ID filter.

The way we're checking for a valid facility ID filter value is by using the existing `PARTIAL_FACILITY_ID: /^\d{4,10}$/,` regex which includes the length requirements of 4-10 digits in the string we're checking.

The facility ID filter value is a string and we're looking to test that an invalid value is rejected, for this case it's a string that is one character short. As we don't have a constant for the minimum number of digits I'm not sure we can do what you're suggesting (if I'm understanding correctly).

We're testing any invalid value is rejected so maybe we could update this const to be alphabetical characters (eg `aaaa`) instead? (as there's unit tests on the validation function itself to test it validates as expected so we could use any invalid case)

Hope that clarifies things, please let me know if not and we can have a chat